import sys
from PyQt6 import QtWidgets, uic
from PyQt6.QtWidgets import QWidget
from PyQt6.QtCore import Qt
from product import ProductApp
from hoadon import OrderManagement # ‚úÖ Import giao di·ªán Qu·∫£n l√Ω S·∫£n ph·∫©m

class MainWindow(QWidget):
    def __init__(self, username, chuc_vu):
        super().__init__()
        uic.loadUi("views/trangchu1.ui", self)

        # Hi·ªÉn th·ªã th√¥ng tin ng∆∞·ªùi d√πng
        self.username = username
        self.chuc_vu = chuc_vu
        self.userName.setText(f"Ng∆∞·ªùi d√πng: {self.username}")
        self.Role.setText(f"Ch·ª©c v·ª•: {self.chuc_vu}")

        # Qu·∫£n l√Ω trang hi·ªÉn th·ªã
        self.trangHienTai = None
        self.danhSachTrang = {}

        # K·∫øt n·ªëi s·ª± ki·ªán c√°c n√∫t
        self.thietLapKetNoi()

        # Hi·ªÉn th·ªã trang ch·ªß m·∫∑c ƒë·ªãnh
        self.hienThiTrangChu()

    def thietLapKetNoi(self):
        """K·∫øt n·ªëi c√°c n√∫t v·ªõi ch·ª©c nƒÉng t∆∞∆°ng ·ª©ng"""
        self.btnProductManagement.clicked.connect(lambda: self.hienThiTrang("sanPham"))
        self.btnInvoiceManagement.clicked.connect(lambda: self.hienThiTrang("hoaDon"))
        self.logoutButton.clicked.connect(self.dangXuat)

    def hienThiTrangChu(self):
        """Hi·ªÉn th·ªã giao di·ªán trang ch·ªß"""
        self.mainContent.setCurrentIndex(0)
        self.trangHienTai = "trangChu"

    def hienThiTrang(self, tenTrang):
        """Chuy·ªÉn ƒë·ªïi hi·ªÉn th·ªã gi·ªØa c√°c trang"""
        if tenTrang not in self.danhSachTrang:
            if tenTrang == "sanPham":
                self.danhSachTrang[tenTrang] = ProductApp()
            elif tenTrang == "hoaDon":
                self.danhSachTrang[tenTrang] = OrderManagement()

        # Th√™m trang v√†o `mainContent`
            self.mainContent.addWidget(self.danhSachTrang[tenTrang])

    # Chuy·ªÉn ƒë·∫øn trang ƒë√£ ch·ªçn
        self.mainContent.setCurrentWidget(self.danhSachTrang[tenTrang])
        self.trangHienTai = tenTrang


    def dangXuat(self):
        """X·ª≠ l√Ω ƒëƒÉng xu·∫•t v√† quay v·ªÅ giao di·ªán ƒëƒÉng nh·∫≠p"""
        from dangnhap import LoginWindow
        self.cuaSoDangNhap = LoginWindow()
        self.cuaSoDangNhap.show()
        self.close()

def main():
    """Ch·∫°y ·ª©ng d·ª•ng"""
    try:
        app = QtWidgets.QApplication(sys.argv)
        cuaSo = MainWindow("admin", "Qu·∫£n l√Ω")  # üõ†Ô∏è Truy·ªÅn th√¥ng tin ƒëƒÉng nh·∫≠p m·∫∑c ƒë·ªãnh
        cuaSo.show()
        sys.exit(app.exec())
    except Exception as e:
        print(f"L·ªói: {str(e)}")

if __name__ == "__main__":
    main()
